api/agent/call.go

```diff
@@ -135,6 +136,17 @@ func buildConfig(app *models.App, route *models.Route) models.Config {
 	return conf
 }
 
+func buildAnnotations(app *models.App, route *models.Route) models.Annotations {
+	ann := make(models.Annotations)
+	for k, v := range app.Annotations {
+		ann[k] = v
+	}
+	for k, v := range route.Annotations {
+		ann[k] = v
+	}
```

Comment:
Annotations in route and app are independent so we don't have a guarantee that keys are unique between route and app that could be a problem with merging them

Reply:
config has the same issue. it's not clear if we should use precedence (route > app), which could be nice in certain cases. e.g. 'app' defaults to something, but 'route' can override and the code doesn't have to change to account for this. or we have separately addressable buckets and the calling code has to figure out how to resolve the conflict themselves. if we do the latter, then it's not much of a stretch to have them put in 2 different keys in 2 places since semantically, it's equivalent to the conflict resolution they'd have to do at the end anyway (this just pushes it up front). I think as long as we're clear about this behavior it's ok (and config).